src/main/java/com/github/cameltooling/lsp/internal/completion/CamelOptionNamesCompletionsFuture.java

```diff
@@ -27,20 +27,23 @@
 import com.github.cameltooling.model.EndpointOptionModel;
 import com.github.cameltooling.model.util.ModelHelper;
 
-public class CamelOptionSchemaCompletionsFuture implements Function<CamelCatalog, List<CompletionItem>>  {
+public class CamelOptionNamesCompletionsFuture implements Function<CamelCatalog, List<CompletionItem>>  {
 
 	private String camelComponentName;
 	private boolean isProducer;
+	private String filterString;
 
-	public CamelOptionSchemaCompletionsFuture(String camelComponentName, boolean isProducer) {
+	public CamelOptionNamesCompletionsFuture(String camelComponentName, boolean isProducer, String filterText) {
 		this.camelComponentName = camelComponentName;
 		this.isProducer = isProducer;
+		this.filterString = filterText;
 	}
 
 	@Override
 	public List<CompletionItem> apply(CamelCatalog catalog) {
 		Stream<EndpointOptionModel> endpointOptions = ModelHelper.generateComponentModel(catalog.componentJSonSchema(camelComponentName), true).getEndpointOptions().stream();
 		return endpointOptions
+				.filter(parameter -> filterString != null && filterString.trim().length()>0 ? parameter.getName().startsWith(filterString) : true)
```

Comment:
this is duplicated code across several classes, might worth extracting a Predicate from it